chain/consensus/mir/manager.go

```diff
@@ -157,11 +157,14 @@ func NewManager(ctx context.Context,
 
 	params := trantor.DefaultParams(initialMembership)
 	params.Iss.SegmentLength = cfg.Consensus.SegmentLength // Segment length determining the checkpoint period.
-	params.Mempool.MaxTransactionsInBatch = 1024
+	if cfg.Consensus.MaxProposeDelay != 0 {
+		params.Iss.MaxProposeDelay = cfg.Consensus.MaxProposeDelay * time.Second
+	}
+	params.Iss.ConfigOffset = cfg.Consensus.ConfigOffset
 	params.Iss.AdjustSpeed(1 * time.Second)
```

Comment:
@matejpavlovic, what is this parameter for?

Reply:
This parameter is about reconfiguration. It says how many Trantor epochs (not to be confused with Filecoin EC epochs) it will take until an agreed-upon configuration will take effect. For example, if `ConfigOffset = 2`, a configuration returned from the call to `NewEpoch` with `nr = 5` will actually be used in Trantor epoch `7` (or maybe `8`, I don't quite remember whether there is an additional offset by 1 related to dealing with old / new epoch numbers).

The only important thing here is, however, that `params.Iss.ConfigOffset` is set to the same value as `cfg.Consensus.ConfigOffset`, so Eudico and Trantor have the same view about which membership is beingused in which epoch.

Reply:
Gotcha, thank you for clarifying. @dnkolegov , we should then add all the checks necessary to ensure that this is the case.